### DISCUSSION TITLE
Firebase config best practice: Where to put in a Nuxt app?

### DISCUSSION THREAD

**Discussion by dosstx** [vuejs/vuefire#1430] [Questions and Help]
Question about best practices when using Nuxt and Nuxt-vuefire:

I have my firebase config in `nuxt.config.ts`, but I also have the same config in a `useFirebase()` composable for using the Firebase JS SDK within various components (ie, login pages, etc). 

Is this good or should I somehow be using runtimeConfig so I am not repeating myself twice?

Update: I saw this: https://github.com/vuejs/vuefire/issues/1411

I assume that's the same thing I'm talking about?

**Comment by posva** [comment]
The config should be in nuxt config. It's accessible via `useRuntimeConfig().public.vuefire` but you shouldn't need it as you have access to the firebase app and that [allows you to use other firebase services](https://vuefire.vuejs.org/guide/other-firebase-services.html#Firebase-Services). What do you need the firebase config for?
